R/s3.get.R

```diff
@@ -1,11 +1,39 @@
 s3.get <- function (bucket, bucket.location = "US", verbose = FALSE, debug = FALSE) {
   AWS.tools:::check.bucket(bucket)
-  x.serialized <- tempfile()
-  s3.cmd <- paste("s3cmd get", bucket, x.serialized, paste("--bucket-location",
-      bucket.location), ifelse(verbose, "--verbose --progress",
-      "--no-progress"), ifelse(debug, "--debug", ""))
-  system(s3.cmd)
-  ans <- readRDS(x.serialized)
-  unlink(x.serialized)
+
+  # Helper function for fetching data from s3
+  fetch <- function(){
+    x.serialized <- tempfile()
+    s3.cmd <- paste("s3cmd get", bucket, x.serialized, paste("--bucket-location",
+    bucket.location), ifelse(verbose, "--verbose --progress",
+    "--no-progress"), ifelse(debug, "--debug", ""))
+    system(s3.cmd)
+
+    ans <- readRDS(x.serialized)
+    unlink(x.serialized)
```

Comment:
Can you move this to just below `x.serialized <- tempfile()` and use

```r
on.exit(unlink(x.serialized), add = TRUE)
```

to unlink even during error?